On PowerPC, unwind-dw2 needs __libc_stack_end to find the AUX
Vector AT_HWCAP. The HWCAP is required to check for the presence of FPU
or Altivec registers.  But `__libc_stack_end' is only exported by ld.so
and ld.so is not included in this link.

This seems to be a gcc-4.4 bug, not reproducible with gcc-4.5. It should
probably be removed at some point.

diff a/dlfcn/Makefile b/dlfcn/Makefile
---
 dlfcn/Makefile |    3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

--- a/dlfcn/Makefile
+++ b/dlfcn/Makefile
@@ -130,7 +130,8 @@
 $(objpfx)bug-atexit2: $(libdl)
 $(objpfx)bug-atexit2.out: $(objpfx)bug-atexit2-lib.so
 
-LDLIBS-bug-atexit3-lib.so = -lstdc++ -lgcc_eh $(common-objpfx)libc_nonshared.a
+LDLIBS-bug-atexit3-lib.so = -lstdc++ -lgcc_eh $(common-objpfx)libc_nonshared.a \
+			    $(elfobjdir)/ld.so
 $(objpfx)bug-atexit3: $(libdl)
 $(objpfx)bug-atexit3.out: $(objpfx)bug-atexit3-lib.so
 
